packages/core/src/components/axes/tool-bar.ts

```diff
@@ -0,0 +1,325 @@
+import settings from "carbon-components/es/globals/js/settings";
+
+export class ToolBar extends Component {
+	type = "tool-bar";
+
+	isZoomBarEnabled: boolean;
+
+	// ul options list element
+	overflowMenuOptions;
+
+	// overflow menu icon rect to control background
+	overflowMenuIcon: any;
+
+	resetZoomMenuItemId = "resetZoomMenuItem";
+
+	// x, y coordinate of overflow menu
+	overflowMenuX = 0;
+	overflowMenuY = 0;
+
+	zoomService = this.services.zoom;
+	// overflow menu icon background padding
+	iconPadding = Configuration.toolBar.iconPadding;
+
+	constructor(model: ChartModel, services: any, configs?: any) {
+		super(model, services, configs);
+
+		this.init();
+	}
+
+	init() {
+		const options = this.model.getOptions();
+
+		// Grab the tooltip element
+		const holder = select(this.services.domUtils.getHolder());
+		const chartprefix = Tools.getProperty(options, "style", "prefix");
+
+		this.overflowMenuOptions = DOMUtils.appendOrSelect(
+			holder,
+			`div.${settings.prefix}--${chartprefix}--overflowMenu`
+		);
+
+		// Apply html content to the tooltip
+		const toolbarTextContainer = DOMUtils.appendOrSelect(
+			this.overflowMenuOptions,
+			"div.content-box"
+		);
+		this.overflowMenuOptions.style("max-width", null);
+
+		// listen to show-tooltip Custom Events to render the tooltip
+		this.services.events.addEventListener(
+			Events.Toolbar.SHOW_MENU_ITEMS,
+			() => {
+				this.overflowMenuOptions.html(this.getOverflowMenuHTML());
+			}
+		);
+
+		// listen to hide-tooltip Custom Events to hide the tooltip
+		this.services.events.addEventListener(
+			Events.Toolbar.HIDE_MENU_ITEMS,
+			() => {
+				this.overflowMenuOptions.html(null);
+			}
+		);
+	}
+
+	render(animate = true) {
+		const isDataLoading = this.zoomService.isDataLoading();
+		this.isZoomBarEnabled =
+			this.zoomService.isZoomBarEnabled() &&
+			!this.zoomService.isEmptyState();
+		// size of overflow menu icon with background
+		const iconSize = Configuration.toolBar.iconSize;
+
+		const svg = this.getContainerSVG();
+		const { width } = DOMUtils.getSVGElementSize(
+			this.services.domUtils.getMainSVG(),
+			{
+				useAttrs: true
+			}
+		);
+
+		// overflow menu option width is 160px
+		// it's set by Carbon component
+		const overflowMenuOptionWidth = 160;
+		// no good solution to get correct ToolBar position
+		// parent x doesn't work well
+		// assume the overflowMenu button has right alignment in layout
+		this.overflowMenuX = width - overflowMenuOptionWidth;
+		this.overflowMenuY =
+			parseFloat(this.parent.node().getAttribute("y")) + iconSize;
+
+		const container = DOMUtils.appendOrSelect(svg, "svg.toolbar-container")
+			.attr("width", "100%")
+			.attr("height", Configuration.toolBar.height)
+			.attr("opacity", 1);
+
+		// clean children first
+		container.html(null);
+		// loading or empty state
+		if (isDataLoading) {
+			// put an empty rect to keep space unchanged
+			DOMUtils.appendOrSelect(container, "svg.toolbar-loading-space")
```

Comment:
```suggestion
			DOMUtils.appendOrSelect(container, "svg.toolbar-loading-spacer")
```

Reply:
Updated in https://github.com/carbon-design-system/carbon-charts/pull/717/commits/bc342d7f9a579d5142cfca38947870f8cd2de4e2.